### DISCUSSION TITLE
Select path in Onedrive to sync

### DISCUSSION THREAD

**Discussion by adsbhatial** [abraunegg/onedrive#1642] [Q&A] [answered]
Is there a option with which I can select with which folder in Onedrive I want to sync. I don't want to completely sync my onedrive account, only a specific folder. So, that whatever is in my  local dir is synced with whatever is in that specific remote dir and not the whole of onedrive.

**Comment by abraunegg** [comment, marked as answer]
@adsbhatial 
Please read https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md - specifically:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-selective-directory-sync
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#selective-sync-via-sync_list-file

**Comment by adsbhatial** [comment]
I apologize but I need a bit more clarification on selective-sync. If I use this, supposedly to only include Backup folder in local dir, then would any of my other folders in my onedrive account would not be touched? Because when I performed a synchronize it deleted everything else in my remote dir which did not match my local onedrive dir. 

My use case is as this:-
I want to synchronize local directory "backup" with remote onedrive dir "backup", if something is deleted in local dir "backup", it should also be deleted in remote dir "backup" but only this directory and should not touch anything else. I want to only upload files which I think is done via upload-only. 

So, if I do, onedrive --synchronize --single-directory-sync 'backups' --upload-only. 
this would satisfy my use case?

Also, If I want the default behavior to be like this, I need to keep a selective-list file, which includes only /Backups. And keep the upload-only to "true" in config. This shouldn't touch my any other remote directories? Is my understanding correct.

**Comment by abraunegg** [comment]
@adsbhatial 
> I'm apologize but I need a bit more clarification on selective-sync. If I use this, supposedly to only include Backup folder in local dir, then would any of my other folders in my onedrive account would not be touched? Because when I performed a synchronize it deleted everything else in my remote dir which did not match my local onedrive dir.

Deletion of data is never good. Make sure you are using the *correct* version - v2.4.13 or greater. If you are using Debian or Ubuntu (or any of the Ubuntu derivatives) - make sure you read and understand https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages ... 

> My use case is as this:-
>I want to synchronize local directory "backup" with remote onedrive dir "backup", if something is deleted in local dir "backup", it should >also be deleted in remote dir "backup" but only this directory and should not touch anything else. I want to only upload files which I >think is done via upload-only.
>
> So, if I do, onedrive --synchronize --single-directory-sync 'backups' --upload-only.
> this would satisfy my use case?

Correct - however ... I would first test with `--dry-run` to ensure this will do what you are wanting. Also read the usage document on `--upload-only`

> Also, If I want the default behavior to be like this, I need to keep a selective-list file, which includes only /Backups. And keep the upload-only to "true" in config. This shouldn't touch my any other remote directories? Is my understanding correct.

Correct .. but also test this with `--dry-run` so that you can confirm the application behavior.

**Reply by adsbhatial** [reply]
Hey, 

Thanks for the clarification. 

I had read the issues in debian's package and removed the symlinks before the synchronize. I'm also using version 2.4.13. The deletion of data was because of my misunderstanding but since I was using this against a test account, there wasn't any issues. 

I've tested both the above cases with single-directory and sync_list and both of them are working as expected
